packages/faceted-search/src/queryClient/tql.js

```diff
@@ -10,13 +10,15 @@ const getBadgeQueryValues = ({ properties }) => [
 const getBadgesQueryValues = badges => badges.map(getBadgeQueryValues);
 
 const isNotEmptyOrNull = value => value && value.length;
+const isValidNumber = value => typeof value === 'number' && !isNaN(value);
```

Comment:
Wouldn't it be easier to fix `isNotEmptyOrNull` ?
Your fix is good for your case, but here we are adding specific cases. It's equivalent to maintain a white list of rules. What about maintaining a black list instead ? And that's what I understand reading `isNotEmptyOrNull`. 
We could, instead of checking `value => value && value.length`
- check if it's not an empty string
- check if it's not undefined
- check if it's not null

Reply:
Thx for review, `isNotEmptyOrNull` is used in two lines :
- https://github.com/Talend/ui/blob/1ad07026e673dc546b80f7ddaefde71560752e49/packages/faceted-search/src/queryClient/tql.js#L21
- https://github.com/Talend/ui/blob/1ad07026e673dc546b80f7ddaefde71560752e49/packages/faceted-search/src/queryClient/tql.js#L15
That's why i didn't fix `isNotEmptyOrNull` after @jmfrancois review.

Reply:
Oh I haven't seen his review. But I still don't understand why checking the id to be different to null, undefined and '' would be ko

Reply:
Why not create a specific method for the id check and the badge value check : 
```suggestion
const isNotEmptyOrNull = value => value && value.length;
const isValidNumber = value => typeof value === 'number' && !isNaN(value);

const isObjectIdNotEmptyOrNull = value => isNotEmptyOrNull(value.id);
const isValidBadge = value => isValidNumber(value) || isNotEmptyOrNull(value);

const filterBadgeWithNoValue = ({ properties }) => {
	if (Array.isArray(properties.value) && properties.value.length) {
		return properties.value.every(isObjectIdNotEmptyOrNull);
	}
	return isValidBadge(properties.value);
};
```

Reply:
@jsomsanith the id is a necessity in the design.
The id is the technical id that will be used in the tql request.
For example : https://github.com/Talend/ui/blob/master/packages/faceted-search/src/queryClient/tql.test.js#L145

Reply:
@nicomaligne Thx, just commit your suggestion 👍

Reply:
I still don't get why checking if a value 
- is not undefined
- is not null
- is not empty string
... is not acceptable for an id.

What I wanted to say with the white list/black list thing is that tomorrow, you have a new type of value, for example a boolean. The check will fail, because the "false" value won't pass the current check.
That's because you have an equivalent of white list, each use case will need to add a new rule.
If you indicates the values that are invalid, as "not provided value", you add a use case, it will still be ok.

That's why I asked if checking the id and the value badge would be ok with the 3 cases (undefined, null, '') described above.